File: ios/objc/StorylyDemo/StorylyDemo/EventHandling/EventHandlingViewController.m
//
//  EventHandlingViewController.m
//  StorylyDemo
//
//  Created by Levent Oral on 8.02.2021.
//  Copyright © 2021 App Samurai Inc. All rights reserved.
//

#import "EventHandlingViewController.h"
#import "Tokens.h"

@interface EventHandlingViewController ()
@property (weak, nonatomic) IBOutlet StorylyView *storylyView;
@end

@implementation EventHandlingViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.storylyView.storylyInit = [[StorylyInit alloc] initWithStorylyId: STORYLY_INSTANCE_TOKEN];
    self.storylyView.rootViewController = self;
    self.storylyView.delegate = self;
}

- (void)storylyLoaded:(StorylyView *)storylyView
       storyGroupList:(NSArray<StoryGroup *> *)storyGroupList {
}

- (void)storylyLoadFailed:(StorylyView *)storylyView
             errorMessage:(NSString *)errorMessage {
}

- (void)storylyActionClicked:(StorylyView *)storylyView
          rootViewController:(UIViewController *)rootViewController
                       story:(Story *)story {
}

- (void)storylyStoryPresented:(StorylyView *)storylyView{
}

- (void)storylyStoryDismissed:(StorylyView *)storylyView {
}

- (void)storylyUserInteracted:(StorylyView *)storylyView
                   storyGroup:(StoryGroup *)storyGroup
                        story:(Story *)story
               storyComponent:(StoryComponent *)storyComponent {
}

- (void)storylyEvent:(StorylyView *)storylyView
               event:(enum StorylyEvent)event
          storyGroup:(StoryGroup *)storyGroup
               story:(Story *)story
      storyComponent:(StoryComponent *)storyComponent {
}

@end

File: ios/swift/StorylyDemo/StorylyDemo/EventHandling/EventHandlingViewController.swift
//
//  EventHandlingViewController.swift
//  StorylyDemo
//
//  Created by Levent Oral on 8.02.2021.
//  Copyright © 2021 App Samurai Inc. All rights reserved.
//

import UIKit
import Storyly

class EventHandlingViewController: UIViewController {
    
    @IBOutlet weak var storylyView: StorylyView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        self.storylyView.storylyInit = StorylyInit(storylyId: STORYLY_INSTANCE_TOKEN)
        self.storylyView.rootViewController = self
        self.storylyView.delegate = self
    }
}

extension EventHandlingViewController: StorylyDelegate {
    func storylyLoaded(_ storylyView: Storyly.StorylyView,
                       storyGroupList: [Storyly.StoryGroup]) {
    }

    func storylyLoadFailed(_ storylyView: Storyly.StorylyView,
                           errorMessage: String) {
    }

    func storylyActionClicked(_ storylyView: Storyly.StorylyView,
                              rootViewController: UIViewController,
                              story: Storyly.Story) {
    }

    func storylyStoryPresented(_ storylyView: Storyly.StorylyView) {
    }

    func storylyStoryDismissed(_ storylyView: Storyly.StorylyView) {
    }

    func storylyUserInteracted(_ storylyView: Storyly.StorylyView,
                               storyGroup: Storyly.StoryGroup,
                               story: Storyly.Story,
                               storyComponent: Storyly.StoryComponent) {
    }

    func storylyEvent(_ storylyView: Storyly.StorylyView,
                      event: Storyly.StorylyEvent,
                      storyGroup: Storyly.StoryGroup?,
                      story: Storyly.Story?,
                      storyComponent: Storyly.StoryComponent?) {
    }
}
